test/functional/wallet_balance.py

```diff
@@ -140,6 +152,48 @@ def run_test(self):
         after = self.nodes[1].getunconfirmedbalance()
         assert_equal(before + Decimal('0.1'), after)
 
+        # Create 3 more wallet txs, where the last is not accepted to the
+        # mempool because it is the third descendant of the tx above
+        for _ in range(3):
+            # Set amount high enough such that all coins are spent by each tx
+            txid = self.nodes[0].sendtoaddress(self.nodes[0].getnewaddress(), 99)
+        assert txid not in self.nodes[0].getrawmempool()
+        assert_equal(self.nodes[0].getbalance(minconf=0), 0)  # wallet txs not in the mempool are untrusted
+
+        self.log.info("Test replacement and reorg of non-mempool tx")
+        tx_orig = self.nodes[0].gettransaction(txid)['hex']
+        # Increase fee by 1 coin
+        tx_replace = tx_orig.replace(
+            struct.pack("<q", 99 * 10**8).hex(),
```

Comment:
😨 is it possible to do this by pre-constructing multiple versions of the same transaction with different fees? My preference would be to avoid doing raw construction byte wrangling if possible.

Reply:
Maybe I should use bumpfee?

Reply:
Perhaps, or the same `createrawtransaction`/`signrawtransaction` method used in `create_transactions()` above.

Just a suggestion. If this adds a ton of additional lines, then it's not worth it. I just find that anything that directly builds/edits transactions can become quite cumbersome to update if the test needs to be changed in future.

Reply:
I couldn't get `create_transactions` to work for this use case after minutes of trying, so I gave up and kept my two-line hack. I think I found two issues with `create_transactions` in the process, so I fixed them up.

Feel free to try it yourself and send  me the commit if you succeed ...

Reply:
No, I agree with you that it's not worth the additional effort. Thanks for trying though!